### PR TITLE
Introduces "mapInteractions" config

### PR DESCRIPTION
**Pull request by KaiVolland**
## Description

This introduces the `mapInteractions` config to the `DefaultApplicationClientConfig`. It allows to configure which interactions should be added to the ol map of the gis-client.

Corresponding handling needs to be added to the shogun-gis-client to actually make this work.

## Pull request type

- [ ] Bugfix
- [x] Feature
- [ ] Dependency updates
- [ ] Tests
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe)

## Do you introduce a breaking change?

- [ ] Yes
- [x] No

## Checklist

- [x] I understand and agree that the changes in this PR will be licensed under the 
  [Apache Licence Version 2.0](https://github.com/terrestris/shogun/blob/main/LICENSE).
- [x] I have followed the [guidelines for contributing](https://github.com/terrestris/shogun/blob/main/CONTRIBUTING.md).
- [x] The proposed change fits to the content of the [code of conduct](https://github.com/terrestris/shogun/blob/main/CODE_OF_CONDUCT.md).
- [ ] I have added or updated tests and documentation, and the test suite passes (run `mvn test` locally).
- [ ] I have added a screenshot/screencast to illustrate the visual output of my update.

### CODE REVIEW

**Review by LukasLohoff — APPROVED** [review]

File: shogun-lib/src/main/java/de/terrestris/shogun/lib/enumeration/MapInteraction.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
+/* SHOGun, https://terrestris.github.io/shogun/
+ *
+ * Copyright © 2020-present terrestris GmbH & Co. KG
```
Comment (dnlkoch) [inline shogun-lib/src/main/java/de/terrestris/shogun/lib/enumeration/MapInteraction.java:3]:
```suggestion
 * Copyright © 2024-present terrestris GmbH & Co. KG
```

**Review by dnlkoch — APPROVED** [review]
Thanks @KaiVolland, really like that idea!

Generally I would favour to have the appropriate client functionality available at the same time since it might confuse users if the configuration (which would be available in the admin immediately) while it won't effect the clients. Is there a linked PR in the shogun-util available right now? If not, I would suggest to wait to merge this one until it's available.

**Comment by KaiVolland** [comment]
Ooops. I'll add the functionality in the client tomorrow. Shouldn't be to hard.

**Comment by dnlkoch** [comment]
No worries, as long as this isn't released, it shouldn't be a problem.

**Pull request merged**